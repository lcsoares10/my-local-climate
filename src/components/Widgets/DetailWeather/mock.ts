

export const mockDetailWheater = {
  clouds: 10,
  pop: 5,
  wind_speed: 100,
  humidity: 0,
  uvi: 70,
};
