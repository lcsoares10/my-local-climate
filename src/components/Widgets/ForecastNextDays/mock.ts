
export default [
    {
        "dt": "20/01/2022",
        "sunrise": 1642666992,
        "sunset": 1642714942,
        "moonrise": 1642722720,
        "moonset": 1642674720,
        "moon_phase": 0.58,
        "temp": {
            "day": 26.22,
            "min": 23.24,
            "max": 28.83,
            "night": 23.91,
            "eve": 28.23,
            "morn": 23.32
        },
        "feels_like": {
            "day": 26.22,
            "night": 24.73,
            "eve": 32.19,
            "morn": 24.11
        },
        "pressure": 1015,
        "humidity": 79,
        "dew_point": 21.94,
        "wind_speed": 5.51,
        "wind_deg": 107,
        "wind_gust": 7.92,
        "weather": [
            {
                "id": 800,
                "main": "Clear",
                "description": "c√©u limpo",
                "icon": "01d"
            }
        ],
        "clouds": 0,
        "pop": 0.46,
        "uvi": 14.65
    },
    {
        "dt": "21/01/2022",
        "sunrise": 1642753435,
        "sunset": 1642801333,
        "moonrise": 1642811220,
        "moonset": 1642764360,
        "moon_phase": 0.62,
        "temp": {
            "day": 25.94,
            "min": 22.89,
            "max": 26.15,
            "night": 23.41,
            "eve": 25.18,
            "morn": 23.04
        },
        "feels_like": {
            "day": 26.68,
            "night": 24.13,
            "eve": 25.87,
            "morn": 23.83
        },
        "pressure": 1014,
        "humidity": 80,
        "dew_point": 22.01,
        "wind_speed": 6.65,
        "wind_deg": 107,
        "wind_gust": 9.47,
        "weather": [
            {
                "id": 802,
                "main": "Clouds",
                "description": "nuvens dispersas",
                "icon": "03d"
            }
        ],
        "clouds": 31,
        "pop": 0.27,
        "uvi": 14.29
    },
 
];
