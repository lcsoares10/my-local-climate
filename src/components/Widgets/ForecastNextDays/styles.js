import styled from "styled-components";

export const ForecastNextDays = styled.div`
  border-top: 1px solid #cecece;
  padding: 20px;
  h4 {
    font-size: ${({ theme }) => theme.fontSizes.large};
    color: ${({ theme }) => theme.colors.white};
    font-weight: 300;
  }
`;

export const WrapperDays = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 5px 20px;

  margin-top: 2rem;
  @media (max-width: ${({ theme }) => theme.media.tablet}) {
    display: none;
  }
`;
