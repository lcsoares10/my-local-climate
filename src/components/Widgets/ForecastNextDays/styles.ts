import styled from "styled-components";

export const ForecastNextDays = styled.div`
  border-top: 3px solid rgba(255, 255, 255, 0.2);
  padding: 20px;


  h4 {
    font-size: ${({ theme }) => theme.fontSizes.large};
    color: ${({ theme }) => theme.colors.white};
    font-weight: 300;
  }

`;

export const WrapperDays = styled.div`
  display: grid;
  align-items: center;
  /* grid-template-columns: repeat(2,150px); */
  grid-template-columns: repeat(auto-fill,300px);
  grid-gap:  1rem;
  margin-top: 2rem;

  @media (max-width: ${({ theme }) => theme.media.mobile}) {
    display: flex;
    overflow-x: scroll;
    padding: 30px;
  }
`;
