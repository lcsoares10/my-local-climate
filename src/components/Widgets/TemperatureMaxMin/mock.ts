export default   {
    day: 20.2,
    eve: 22.5,
    max: 10.6,
    min: 23.6,
    morn: 25.9,
    night: 12.1,
  };