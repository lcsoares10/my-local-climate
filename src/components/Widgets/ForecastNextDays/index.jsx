import React from "react";
import * as S from "./styles";
import rain from "../../../../public/rain.gif";
import Image from "next/image";

import WheaterCard from "../WeatherCard";

function ForecastNextDays() {
  const dataTest = [1, 2, 3, 4];
  return (
    <S.ForecastNextDays>
      <h4>Previs√£o para os proximos 4 dias</h4>

      <S.WrapperDays>
        {dataTest.map((day) => (
          <WheaterCard  hoverEffect/>
        ))}
      </S.WrapperDays>
    </S.ForecastNextDays>
  );
}

export default ForecastNextDays;
