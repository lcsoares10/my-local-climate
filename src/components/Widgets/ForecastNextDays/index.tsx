import useWeather from "../../../hooks/useWeather";
import WheaterCard from "../WeatherCard";
import { TDailyWeatherFormat } from "../../../context/weatherProvider";
import SkeletonForecastNextDays from "./skeleton";

import * as S from "./styles";

type TForecastNextDays = {
  weatherDaily: Array<TDailyWeatherFormat>;
};

const ForecastNextDays: React.FC<TForecastNextDays> = ({
  children,
  weatherDaily,
  ...props
}) => {
  const { setSelectedDay, selectedDay } = useWeather();

  if (!weatherDaily) {
    return (
      <S.ForecastNextDays {...props}>
        <SkeletonForecastNextDays />
      </S.ForecastNextDays>
    );
  }

  return (
    <S.ForecastNextDays {...props}>
      {children}

      <S.WrapperDays>
        {weatherDaily &&
          weatherDaily.map((day, key) => (
            <div
              key={"WheaterCard_" + key}
              onClick={() => setSelectedDay()}
              data-testid={"wheater-card-container"}
            >
              <WheaterCard
                weather={day.weather[0]}
                temp={day.temp.day}
                dt={day.dt}
                hoverEffect
                selected={day.dt === selectedDay}
              />
            </div>
          ))}
      </S.WrapperDays>
    </S.ForecastNextDays>
  );
};

export default ForecastNextDays;
