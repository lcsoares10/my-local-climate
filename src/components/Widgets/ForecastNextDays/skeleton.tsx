import { SkeletonWheaterCard } from "../WeatherCard/skeleton";
import * as S from "./styles";

const SkeletonForecastNextDays = () => (
  <S.WrapperDays data-testid="skeleton-wrapper-days">
    <SkeletonWheaterCard />
    <SkeletonWheaterCard />
    <SkeletonWheaterCard />
    <SkeletonWheaterCard />
  </S.WrapperDays>
);
export default SkeletonForecastNextDays;
