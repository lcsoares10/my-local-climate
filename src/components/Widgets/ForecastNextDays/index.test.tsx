import { fireEvent, render, screen } from "@testing-library/react";
import Theme from "../../../theme";
import ForecastNextDays from ".";
import mock from "./mock";
import userEvent from "@testing-library/user-event";
import  WeatherCard from "../WeatherCard"

describe("<ForecastNextDays />", () => {
  it("should view in display skeleton", async () => {
    render(
      <Theme>
        <ForecastNextDays weatherDaily={null} />
      </Theme>
    );

    expect(screen.getByTestId("skeleton-wrapper-days")).toBeTruthy();
  });

  it("should view a title and two cards ", async () => {


    const {container}=render(
      <Theme>
        <ForecastNextDays weatherDaily={mock} >
          <h4>Previsão para os proximos dias</h4>
        </ForecastNextDays>
      </Theme>
    );
    expect(
      screen.getByRole("heading", { name: /Previsão para os proximos dias/i })
    ).toBeInTheDocument();
    expect(screen.getAllByTestId("wheater-card-container")).toHaveLength(2);

    expect(container.firstChild).toMatchSnapshot();

    // const wheaterCard = screen.getAllByTestId("wheater-card-container");
    // //wheaterCard.forEach((card) => userEvent.click(card));

    // //expect(jest.spyOn(ForecastNextDays, 'teste')).toHaveBeenCalledWith("1");
  });


});
