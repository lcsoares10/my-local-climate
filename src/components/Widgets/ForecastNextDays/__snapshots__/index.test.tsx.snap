// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastNextDays /> should view in display skeleton 1`] = `
<div
  class="sc-bdvvtL khGXNt"
  id="button_theme"
  type="light"
>
  <svg
    aria-hidden="true"
    class="StyledIconBase-ea9ulj-0 bsimgx"
    color="rgb(38,38,38)"
    fill="currentColor"
    focusable="false"
    height="30"
    viewBox="0 0 24 24"
    width="30"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="none"
      height="24"
      width="24"
    />
    <path
      d="M12 3a9 9 0 109 9c0-.46-.04-.92-.1-1.36a5.389 5.389 0 01-4.4 2.26 5.403 5.403 0 01-3.14-9.8c-.44-.06-.9-.1-1.36-.1z"
    />
  </svg>
</div>
`;
